common/core/hwclayer.cpp

```diff
@@ -325,25 +325,69 @@ bool HwcLayer::IsCursorLayer() const {
   return is_cursor_layer_;
 }
 
+HwcRect<int> HwcLayer::TransformDamage() {
+  HwcRect<int> ret;
+  int ox = 0, oy = 0;
+  HwcRect<int> tr_surface_damage =
```

Comment:
instead of allocating temp rects, u can use current_Rendering_Damage_ ?

Reply:
I think that would impact the readability of the code? 
I was thinking of removing return value and just assign the ret rect to current_rendering_damage.

Reply:
hmm no. You can move this chunk of code to the calling function itself I guess to keep it all in one place.

Reply:
okay.